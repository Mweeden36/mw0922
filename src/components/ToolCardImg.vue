<template>
  <div class="tool-card-img__wrapper">
    <img v-if="!isError" :src="imgSrc" @error="onImageError" />
    <vue-feather v-else type="tool" />
  </div>
</template>

<script>
export default {
  name: "ToolCardImg",

  props: {
    code: {
      type: String,
      required: true,
    },
  },

  data() {
    return {
      isError: false,
    };
  },

  computed: {
    imgSrc() {
      return `../assets/images/${this.code}.jpg`;
    },
  },

  methods: {
    onImageError() {
      this.isError = true;
    },
  },
};
</script>

<style lang="scss" scoped>
@import "../styles/colors";
@import "../styles/borders";

.tool-card-img__wrapper {
  padding: 10px;
  border: 1px solid $color-tertiary;
  border-radius: $border-radius;
  box-shadow: $box-shadow;
  background: #fff;
}
</style>
