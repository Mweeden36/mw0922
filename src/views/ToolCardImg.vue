<script setup>
import { computed, ref } from 'vue';

const props = defineProps({
  code: {
    type: String,
    required: true,
  },
});

let isError = ref(false);

const imgSrc = computed(() => {
  return `../assets/images/${props.code}.jpg`
});

const onImageError = () => {
  isError.value = true;
};
</script>

<template>
  <div class="tool-card-img__wrapper">
    <img v-if="!isError" :src="imgSrc" @error="onImageError" />
    <vue-feather v-else type="tool" />
  </div>
</template>

<style lang="scss" scoped>
@import '../styles/colors';
@import '../styles/borders';

.tool-card-img__wrapper {
  padding: 10px;
  border: 1px solid $color-tertiary;
  border-radius: $border-radius;
  box-shadow: $box-shadow;
}
</style>