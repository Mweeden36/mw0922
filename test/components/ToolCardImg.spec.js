import ToolCardImg from "../../src/components/ToolCardImg.vue";
import VueFeather from "vue-feather";
import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

describe("Toolcard.vue", () => {
  it("should compute the correct image source url", () => {
    const { vm } = mount(ToolCardImg, {
      propsData: {
        code: "JAKD",
      },
      global: {
        stubs: {
          "vue-feather": VueFeather,
        },
      },
    });
    expect(vm.imgSrc).toEqual("/images/JAKD.jpg");
  });

  it("should generate the correct non-error html", () => {
    const wrapper = mount(ToolCardImg, {
      propsData: {
        code: "JAKD",
      },
      global: {
        stubs: {
          "vue-feather": VueFeather,
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should generate the correct non-error html", () => {
    const wrapper = mount(ToolCardImg, {
      propsData: {
        code: "JAKD",
      },
      global: {
        stubs: {
          "vue-feather": VueFeather,
        },
      },
    });

    expect(wrapper.vm.isError).toEqual(false);
    wrapper.vm.onImageError();
    expect(wrapper.vm.isError).toEqual(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
