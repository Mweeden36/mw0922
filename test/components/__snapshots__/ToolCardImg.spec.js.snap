// Vitest Snapshot v1

exports[`Toolcard.vue > should generate the correct non-error html 1`] = `"<div class=\\"tool-card-img__wrapper\\" data-v-40df69f9=\\"\\"><img src=\\"/images/JAKD.jpg\\" data-v-40df69f9=\\"\\"></div>"`;

exports[`Toolcard.vue > should generate the correct non-error html 2`] = `"<div class=\\"tool-card-img__wrapper\\" data-v-40df69f9=\\"\\"><img src=\\"/images/JAKD.jpg\\" data-v-40df69f9=\\"\\"></div>"`;
